getters.go

```diff
@@ -0,0 +1,73 @@
+package duty
+
+import (
+	"errors"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/duty/models"
+	"github.com/patrickmn/go-cache"
+	"gorm.io/gorm"
+)
+
+// getterCache caches the results for all the getters in this file.
+var getterCache = cache.New(time.Second*30, time.Minute*5)
+
+type cachedItem struct {
+	resource any
+	notFound bool
+}
+
+func FindCachedAgent(ctx dbContext, agentID string) (*models.Agent, error) {
```

Comment:
Can we also add for Person, Component and Config?

Why is the first parameter a private struct ?

Reply:
added Person, Component and Config.

The first parameters is an interface.

Reply:
Yes, but why is it private ?

Reply:
I don't see why that should be exported.

Reply:
We should also handle the special case of all 000-s agent and return nil

Reply:
> I don't see why that should be exported. 

The types of the parameters for exported function should all be exported, otherwise you may not be able to call the function, or the type info gets lost in docs etc..

Reply:
As interfaces are implicit, callers outside of the package will still be able to call the function. And the callers can still see what needs to be defined to satisfy the interface.

From a usability perspective, I still do not see the difference. Moreover, exporting it also enables other importers to internally use it _(eg: config-db using it internally without any calls to duty)_. Any future change to `dbContext` might cause the consumers to break.

I would lean more towards minimizing the export footprint of the package and only exporting the must-haves.

The only downside I see is that it might not be well documented with `go doc`.